python/run-tests

```diff
@@ -1,6 +1,20 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+FWDIR="$(cd "`dirname $0`"; pwd)"
+cd "$FWDIR"
+
 set -x
 if [ -z "$SPARK_HOME" ]; then
    echo "$(tput setaf 1)Error: SPARK_HOME is not set, may not be able to run tests.$(tput sgr0)"
 fi
-PYTHON_PATH=./ nosetests --logging-level=INFO --detailed-errors --verbosity=2 --with-coverage --cover-html-dir=./htmlcov --cover-package=sparktestingbase --traverse-namespace 
+PYTHON_PATH=./ nosetests \
+    --logging-level=INFO \
+    --detailed-errors \
+    --verbosity=2 \
+    --with-coverage \
+    --cover-branches \
```

Comment:
This was a nice excuse to play around with Codecov's new support for Python branch coverage.